File: oolab/src/main/java/agh/ics/oop/model/MapDirection.java
package agh.ics.oop.model;

public enum MapDirection {
    NORTH(new Vector2d(0,1)),
    SOUTH(new Vector2d(0,-1)),
    WEST(new Vector2d(-1,0)),
    EAST(new Vector2d(1,0));

    private final Vector2d direction;

    MapDirection(Vector2d vector){
        direction = vector;
    }

    @Override
    public String toString(){
        return switch(this) {
            case NORTH -> "Północ";
            case SOUTH -> "Południe";
            case WEST -> "Zachód";
            case EAST -> "Wschód";
        };
    }

    protected MapDirection next(){
        return switch(this) {
            case NORTH -> MapDirection.EAST;
            case SOUTH -> MapDirection.WEST;
            case WEST -> MapDirection.NORTH;
            case EAST -> MapDirection.SOUTH;
        };
    }

    protected MapDirection previous(){
        return switch(this) {
            case NORTH -> MapDirection.WEST;
            case SOUTH -> MapDirection.EAST;
            case WEST -> MapDirection.SOUTH;
            case EAST -> MapDirection.NORTH;
        };
    }

    private Vector2d toUnitVector(){
        return direction;
    }
}

File: ProjektKotlina/src/main/kotlin/model/MapDirection.kt
package model

enum class MapDirection(val direction: Vector2d) {
    NORTH(Vector2d(0, 1)),
    SOUTH(Vector2d(0, -1)),
    WEST(Vector2d(-1, 0)),
    EAST(Vector2d(1, 0));

    override fun toString(): String = when (this) {
        NORTH -> "Północ"
        SOUTH -> "Południe"
        WEST -> "Zachód"
        EAST -> "Wschód"
    }

    fun next(): MapDirection = when (this) {
        NORTH -> EAST
        SOUTH -> WEST
        WEST -> NORTH
        EAST -> SOUTH
    }

    fun previous(): MapDirection = when (this) {
        NORTH -> WEST
        SOUTH -> EAST
        WEST -> SOUTH
        EAST -> NORTH
    }

}
